### PR TITLE
Tracking clicks on email suggestion

### PR DESCRIPTION
**Pull request by MathieuGilbert**
- Refactored mailchecker code into a class, following coffeescript standards,  and in own folder
- Adding event to the dataLayer when the domain suggestion link is clicked on, only if the consuming app has dataLayer defined
- Renamed *.js.coffee to *.coffee

:elephant:

### CODE REVIEW

File: app/assets/javascripts/ama_layout/mailcheck/autocorrect_email.coffee [hunk 1]
```diff
@@ -0,0 +1,81 @@
+class AMALayout.EmailSuggestion
```
Comment (adamrmelnyk) [inline app/assets/javascripts/ama_layout/mailcheck/autocorrect_email.coffee]: Did you mv this instead of git mv?

Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/mailcheck/autocorrect_email.coffee]: Nope, not sure why it looks like it... from history:
image

File: app/assets/javascripts/ama_layout/mailcheck/autocorrect_email.coffee [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          if !$('.email_hint').length
+            $("<div class='email_hint'>#{text}</div>").insertAfter(element).fadeIn 150
+          else
+            $('.email_hint').html text
+        empty: (element) ->
+          $('.email_hint').html ''
+
+    $(document).on 'click', '.email_hint .suggestion a.email_domain', =>
+      @trackUsage()
+      email_hint = $(this).parents('.email_hint')
```
Comment (MathieuGilbert) [inline app/assets/javascripts/ama_layout/mailcheck/autocorrect_email.coffee]: This `this` is wrong, since it changed from `->` to `=>`. Fixing...

**Comment by jessedoyle** [comment]
🐘 🐘 🐘

**Comment by darkodosenovic** [comment]
:elephant: :elephant: :elephant:

**Pull request merged**